olive/evaluator/olive_evaluator.py

```diff
@@ -128,6 +129,10 @@ def evaluate(
     ) -> MetricResult:
         metrics_res = {}
         for metric in metrics:
+            # use model io_config if user does not specify input_names and input_shapes
+            # only do this if data_config or dataloader is not provided
+            # priority: dataloader_func > data_config > user_config.input_names/input_shapes > model io_config
```

Comment:
@jambayk , please make sure the comment is right since we change the priority order you previous added.

Reply:
Yes, this new priority looks good to me.